src/nodes/dir.rs

```diff
@@ -37,46 +57,43 @@ struct MutableDir {
     parent: u64,
     underlying_path: Option<PathBuf>,
     attr: fuse::FileAttr,
-    children: HashMap<OsString, Arc<Node>>,
+    children: HashMap<OsString, Dirent>,
 }
 
 impl Dir {
-    /// Creates a new scaffold directory to represent the root directory.
-    ///
-    /// `time` is the timestamp to be used for all node times.
+    /// Creates a new scaffold directory to represent an in-memory directory.
     ///
-    /// `uid` and `gid` indicate the ownership details of the node.  These should always match the
-    /// values of the currently-running process -- but not necessarily if we want to let users
-    /// customize these via flags at some point.
-    pub fn new_root(time: Timespec, uid: unistd::Uid, gid: unistd::Gid) -> Arc<Node> {
-        let inode = fuse::FUSE_ROOT_ID;
+    /// The directory's timestamps are all set to the current time, and the ownership is set to the
+    /// current user.
+    pub fn new_empty(inode: u64, parent: Option<&Node>) -> Arc<Node> {
```

Comment:
You'll find it easier to test if you take `now` as an argument.

Reply:
The same would apply to the UID/GID, but I guess those are less of a bother for the testing case. (All of them should be passed as arguments but I went for "simplicity".) Anyway, plumbed `now` as an argument.